Sources/Macros/SampleBuilder/SampleBuilderMacro+Class.swift

```diff
@@ -0,0 +1,200 @@
+/*
+ This source file is part of SwiftAndTipsMacros
+ 
+ Copyright (c) 2023 Pedro Rojas and project authors
+ Licensed under MIT License
+ */
+//
+//  SampleBuilderMacro+Class.swift
+//
+//
+//  Created by Philip Eram on 10/26/23.
+//
+
+import SwiftSyntax
+import DataGenerator
+import DataCategory
+import SwiftSyntaxMacros
+
+extension SampleBuilderMacro {
+    static func SampleBuilderMacroForClass(
+        classDecl: ClassDeclSyntax,
+        numberOfItems: Int,
+        generatorType: DataGeneratorType,
+        context: MacroExpansionContext
+    ) -> [SwiftSyntax.DeclSyntax] {
+        let validParameters = getValidParameterList(
+            from: classDecl,
+            generatorType: generatorType,
+            context: context)
+        
+        let sampleCode = generateSampleCodeSyntax(
+            sampleData: generateSampleData(
+                parameters: validParameters,
+                numberOfItems: numberOfItems,
+                generatorType: generatorType
+            )
+        )
+        
+        return [DeclSyntax(sampleCode)]
+    }
+    
+    static func getValidParameterList(
+        from classDecl: ClassDeclSyntax,
+        generatorType: DataGeneratorType,
+        context: MacroExpansionContext
+    ) -> [ParameterItem] {
+        let storedPropertyMembers = classDecl.memberBlock.members
```

Comment:
I think this `classDecl` is the only difference compared to the struct implementation. Since both classes and structs are pretty similar in terms of syntax, Should we refactor this code to make it available for both?